vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java

```diff
@@ -241,6 +243,20 @@ private boolean compareNumber(Object value, Object compareValue, String fieldNam
   private boolean compareArray(Object value, Object compareValue, String fieldName) {
     if (!compareValue.getClass().isArray())
       throw new MappingException("Contents are not equal: " + fieldName);
+    int l1 = Array.getLength(value);
+    int l2 = Array.getLength(compareValue);
+    if (l1 != l2) {
+      throw new MappingException("different lenght in arrays: " + l1 + " / " + l2);
+    }
+    // first clear wether all elements are existing, equal the order
```

Comment:
just a special check to guarantee the detailed information wether an array is different in length, contains all elements of the other one and is sorted correctly in this order. Arrays.equals wouldn't give the second part.